uaa_client/authentication.py

```diff
@@ -93,13 +93,21 @@ def get_user_by_email(email):
         of security policies for logins.
         '''
 
+        APPROVED_DOMAINS = getattr(settings, 'UAA_APPROVED_DOMAINS', [])
+
         try:
             return User.objects.get(email__iexact=email)
         except User.DoesNotExist:
-            logger.info(
-                'User with email {} does not exist'.format(email)
-            )
-            return None
+            email_pieces = email.split('@')
+            if email_pieces[1] in APPROVED_DOMAINS:
+                User.objects.create_user(email_pieces[0], email)
```

Comment:
Given all the potential edge cases, I wonder if it might be useful to split out this logic into a separate method, so that `get_user_by_email()` doesn't get too long, and so that the logic can be easily tested independently. This might also make it easier for subclasses to leverage functionality from this class without having to copy/paste a bunch of common logic, maybe.